scripts/common/proxy.sh

```diff
@@ -1,51 +1,91 @@
         ENV_NO_PROXY="$NO_PROXY"
     elif [ -n "$no_proxy" ]; then
         ENV_NO_PROXY="$no_proxy"
     fi
 
-    # Need to peek at the yaml spec to find if a proxy is needed to download the util binaries
+    # here we sanitize the proxy configuration. we make sure that both upper
+    # and lower case variables point to the same value.
+    if [ -n "$ENV_HTTP_PROXY_ADDRESS" ]; then
+        export http_proxy="$ENV_HTTP_PROXY_ADDRESS"
+        export HTTP_PROXY="$ENV_HTTP_PROXY_ADDRESS"
+    fi
+    if [ -n "$ENV_HTTPS_PROXY_ADDRESS" ]; then
+        export https_proxy="$ENV_HTTPS_PROXY_ADDRESS"
+        export HTTPS_PROXY="$ENV_HTTPS_PROXY_ADDRESS"
+    fi
+    if [ -n "$ENV_NO_PROXY" ]; then
+        export no_proxy="$ENV_NO_PROXY"
+        export NO_PROXY="$ENV_NO_PROXY"
+    fi
+
+    # if proxy is configured we need to make sure that kubectl can reach
+    # the apiserver without going through the proxy.
+    if [ -n "$ENV_HTTP_PROXY_ADDRESS" ] || [ -n "$ENV_HTTPS_PROXY_ADDRESS" ]; then
+        kubectl_no_proxy
+    fi
+}
+
+# proxy_bootstrap reads the proxy configuration from the environment and
+# overrides them with the configuration provided by the user through the
+# installer yaml. at the end of this process three variables are set:
+# ENV_HTTP_PROXY_ADDRESS, ENV_HTTPS_PROXY_ADDRESS, and ENV_NO_PROXY.
+function proxy_bootstrap() {
+    # read and sanitize proxy configuration from environment variables.
+    read_proxy_config_from_env
+
+    # users can still provide a different proxy by patching the installer
+    # yaml, we need to verify if this is the case and then use the proxy
+    # set in the yaml for both http and https.
     if [ -n "$INSTALLER_SPEC_FILE" ]; then
         local overrideProxy=$(grep "proxyAddress:" "$INSTALLER_SPEC_FILE" | grep -o "http[^'\" ]*")
         if [ -n "$overrideProxy" ]; then
+            ENV_HTTP_PROXY_ADDRESS="$overrideProxy"
+            ENV_HTTPS_PROXY_ADDRESS="$overrideProxy"
+            export http_proxy="$overrideProxy"
             export https_proxy="$overrideProxy"
-            kubectl_no_proxy
+            export HTTP_PROXY="$overrideProxy"
+            export HTTPS_PROXY="$overrideProxy"
             echo "Bootstrapped proxy address from installer spec file: $https_proxy"
             return
         fi
     fi
-    local proxy=$(echo "$INSTALLER_YAML" | grep "proxyAddress:" | grep -o "http[^'\" ]*")
-    if [ -n "$proxy" ]; then
-        export https_proxy="$proxy"
-        kubectl_no_proxy
-        echo "Bootstrapped proxy address from installer yaml: $https_proxy"
-        return
-    fi
 
-    if [ -n "$ENV_PROXY_ADDRESS" ]; then
-        export https_proxy="$ENV_PROXY_ADDRESS"
-        kubectl_no_proxy
-        log "Bootstrapped proxy address from ENV_PROXY_ADDRESS: $https_proxy"
+    local proxy=$(echo "$INSTALLER_YAML" | grep "proxyAddress:" | grep -o "http[^'\" ]*")
+    if [ -z "$proxy" ]; then
         return
     fi
+    ENV_HTTP_PROXY_ADDRESS="$proxy"
```

Comment:
im worried that you no longer call `kubectl_no_proxy` in this case

Reply:
This comment is similar to this [other](https://github.com/replicatedhq/kURL/pull/4538#discussion_r1210310437) comment on the same review. I will resolve this one and we can discuss on the first one.